### ISSUE TITLE
If "Max Results Per Quality" has a value, sometimes no sources (with a stated Quality) are returned

### ISSUE THREAD

**Issue by P-Jay357** [TheBeastLT/torrentio-scraper#186]
I've had this issue with Star Trek: Prodigy.

If any value is set for "Max Results Per Quality" when configuring the Addon, then the episodes for this TV show have either 0 sources, or 1 "[RD+] Torrentio" source (with no stated quality). If I clear the value for "Max Results Per Quality", then I can see the 10+ sources for each episode.

I've not seen this issue occur for any of the other TV shows I've watched using Torrentio.

**Comment by TheBeastLT** [comment]
Everything is working as expected. For some qualities probably the top links are not cached and you have disabled download to torrentio links.

**Comment by P-Jay357** [comment]
I don't think this behaviour is expected for my use case though. If it was a cache issue, then why do the links get returned fine with no max per quality set, but providing a max value returns either 0 links or just the 1 here where no quality is set for it (it doesn't say 1080p, etc.)? The only thing I changed between these 2 scenarios is the "Max results per quality" value.

Because as we can see there are more than 3 1080p links returned when no max value is set, shouldn't it still return the top 3 1080p links when a max value is present? This issue has only happened for Star Trek: Prodigy. Every other TV show or movie works fine with the 3 max per quality value, so it makes me think it's a weird edge-case.

![TorrentioConfigNoMax](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/c0225bec-b086-4755-9a18-8fce8ddb56d2)
![ResultsWithNoMax](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/50137400-3be6-4b1d-b21a-70f5d0afb931)
![TorrentioCOnfigWithMax](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/856012af-e6c3-472e-bc3a-b81555c4b3c6)
![ResultsWithMax](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/2a30723c-4ad4-42cb-8df1-2084d97c158e)

**Comment by TheBeastLT** [comment]
Because you're sorting by quality then size, when you limit to 3 options, there are other torrents have larger size thus they are selected over others links, but since you've disabled showing download links you're not seeing those 2entries as they are not cached. If you'd remove the checkmark from the don't show download to debrid you'd see them.

**Comment by P-Jay357** [comment]
Thanks for replying! I've set the "Max Results Per Quality" value to 3 and have un-checked the "Don't show download to debrid links" option. None of the above 1080p (or any other cached debrid link with a displayed quality) links are shown though. Like I said, this has only happened to me for Star Trek: Prodigy. All other TV shows and Movies work as expected (showing top 3 4k, 1080p, 720p, etc.) links. I think this may be a weird edge-case, as with the options I have set in these screenshots there should be some 1080p links showing. If you have a debrid service, have you been able to replicate this issue on your end?

![TorrentioSettupWithShowDownloadLinks](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/401d17ca-8cd6-4234-b3bd-fa3aa4b407c7)

![StremioResultsWithShowDownloadLinks](https://github.com/TheBeastLT/torrentio-scraper/assets/43815586/8089459a-6618-4bab-9a3e-fe806e1c48a9)
